android/services/mongodb-remote/src/main/java/com/mongodb/stitch/android/services/mongodb/remote/RemoteMongoCollection.java

```diff
@@ -474,17 +475,43 @@
   /**
    * Watches specified IDs in a collection.  This convenience overload supports the use case
    * of non-{@link BsonValue} instances of {@link ObjectId}.
+   *
    * @param ids unique object identifiers of the IDs to watch.
    * @return the stream of change events.
    */
-  Task<ChangeStream<Task<ChangeEvent<DocumentT>>, DocumentT>> watch(final ObjectId... ids);
+  Task<ChangeStream<Task<ChangeEvent<DocumentT>>>> watch(final ObjectId... ids);
 
   /**
    * Watches specified IDs in a collection.
+   *
+   * @param ids the ids to watch.
+   * @return the stream of change events.
+   */
+  Task<ChangeStream<Task<ChangeEvent<DocumentT>>>> watch(final BsonValue... ids);
+
+  /**
+   * Watches specified IDs in a collection.  This convenience overload supports the use case
+   * of non-{@link BsonValue} instances of {@link ObjectId}. This convenience overload supports the
+   * use case of non-{@link BsonValue} instances of {@link ObjectId}. Requests a stream where the
+   * full document of update events, and several other unnecessary fields are omitted from the
+   * change event objects returned by the server. This can save on network usage when watching
+   * large documents.
+   *
+   * @param ids unique object identifiers of the IDs to watch.
+   * @return the stream of change events.
+   */
+  Task<ChangeStream<Task<CompactChangeEvent<DocumentT>>>> watchCompact(final ObjectId... ids);
+
```

Comment:
I know I'm late to this conversation but it seems like we are sacrificing usability for not breaking the current API. Theoretically, why can't we make ChangeEvent<DocumentT> an interface and return it in both of these methods? 

Also is it better to have another name for the method "watchCompact" as opposed to an overload of the watch method that is: watch(final bool fullDocument = false, final ObjectId... ids)

Reply:
We can't make `ChangeEvent` an interface because the two change event models are fundamentally different. 

For that same reason, we have multiple methods. You can't overload on return type in Java, so if you have a different return type depending on the argument, you either have to have a generic return type which is not very user friendly, or we'd have to have this be a stream of `BaseChangeEvent`, which is not very useful to users since they would have to know to typecast to the type of change event they wany. Happy to clarify offline as well.